### PR TITLE
#161967003 Migrate to a Postgres Database

### PR DESCRIPTION
**Pull request by gitau-kinuthia**
#### What does this PR do?

Adds Postgres database integration

#### Description of Task to be completed?

- Integrate a Postgres database and make migrations.
- stop tracking  of Django migrations by adding all migration folders to .gitignore
- update requrements.txt with Django-environ and psycopg2
- add .env file with local database configuration
- add database settings in settings.py
- add unit tests for database connection and vendor verification

#### How should this be manually tested?
```
1. git clone https://github.com/andela/ah-jumanji.git
2. cd ah-jumanji
3. git checkout ch-configure-postgres-db-161967003
4.  Setup a `virtual environment` and activate it
5. pip install -r requirements.txt
6. touch .env
7. echo "DATABASE_URL=postgres://username:password@lhost/db_name" >> .env
8. python manage.py migrate
9. pytest -k "TestDatabaseConnection"  
```
#### Any background context you want to provide?
This PR attempts to implement a [12-factor application](https://www.12factor.net/) by adding an external config file. This is for a cleaner and scalable codebase

#### What are the relevant pivotal tracker stories?
[#161967003](https://www.pivotaltracker.com/story/show/161967003)

#### Screenshots (if appropriate)
screenshot 2018-12-03 at 14 34 20

#### Checklist:

- [x] My code follows the style guidelines of this project
- [x] At least 2 people have reviewed my PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] My PR has one commit.

### CODE REVIEW

File: authors/apps/core/tests/test_db_connection.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import logging
```
Comment (ghost) [inline authors/apps/core/tests/test_db_connection.py]: This looks like an unused `import` ...

**Review by mashafrancis — APPROVED** [review]

**Review by mashafrancis — APPROVED** [review]
The integration looks fine with me.

File: authors/apps/core/tests/test_db_connection.py (hunk 1 above)
Comment (gitau-kinuthia) [reply authors/apps/core/tests/test_db_connection.py]: @dmithamo1 the unused import has been noted and removed. Thanks for the feedback

**Review by ghost — APPROVED** [review]
This looks ok.
Except it would be great if we all got used to using this PR template. (Include the checklist at the bottom, and `place an x within the square brackets - no spaces - to place a tick on an item that you've satisfied)
https://raw.githubusercontent.com/andela/ah-jumanji/develop/.github/jumanji_pull_request_template.md

**Review by fabzer0 — APPROVED** [review]

**Review by gwako94 — APPROVED** [review]

**Review by GransonO — APPROVED** [review]
The work looks good. No merge conflicts and all tests are passing as expected.
`LGTM`

**Review by ghost — COMMENTED** [review]
1. The set up process as defined in the `PR` is different from that in the `README`. Why this difference?
2. I cannot get the tests to run in my local machine, despite following your set up instructions exactly as defined. 

ANSWER
The documentation is written to apply to the branch develop hence the additional statemnts that are more significant there
With regards to the second remark, make sure you use an appropriate DATABASE_URL ssuch as `postgres://username:password@localhost/database_name` . The database and the localhost should be already running on your machine

**Review by ghost — CHANGES_REQUESTED** [review]
1. The set up process as defined in the `PR` is different from that in the `README`. Why this difference?
2. I cannot get the tests to run in my local machine, despite following your set up instructions exactly as defined.

**Review by ghost — APPROVED** [review]

**Review by EmmanuelBeja — APPROVED** [review]

**Pull request merged**